# From https://github.com/facebook/zstd/pull/1916
#
diff -ur zstd-1.5.7/lib/Makefile zstd-1.5.7-new/lib/Makefile
--- zstd-1.5.7/lib/Makefile	2025-02-19 00:04:24.000000000 +0000
+++ zstd-1.5.7-new/lib/Makefile	2026-03-11 13:23:45.442071338 +0000
@@ -286,6 +286,7 @@
 LIBDIR      ?= $(libdir)
 includedir  ?= $(PREFIX)/include
 INCLUDEDIR  ?= $(includedir)
+DLLDIR      ?= $(PREFIX)/bin
 
 PCINCDIR := $(patsubst $(PREFIX)%,%,$(INCLUDEDIR))
 PCLIBDIR := $(patsubst $(EXEC_PREFIX)%,%,$(LIBDIR))
@@ -363,8 +364,13 @@
 	[ -e $(DESTDIR)$(LIBDIR) ] || $(INSTALL) -d -m 755 $(DESTDIR)$(LIBDIR)/
 	@echo Installing shared library
 	$(INSTALL_PROGRAM) $(LIBZSTD) $(DESTDIR)$(LIBDIR)
+ifneq (,$(filter MSYS% MINGW%,$(UNAME)))
+	@$(INSTALL) -d -m 755 $(DESTDIR)$(DLLDIR)/
+	@$(INSTALL_PROGRAM) $(LIBZSTD) $(DESTDIR)$(DLLDIR)
+else
 	ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR)
 	ln -sf $(LIBZSTD) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
+endif
 
 .PHONY: install-includes
 install-includes:
@@ -377,8 +383,13 @@
 .PHONY: uninstall
 uninstall:
 	$(RM) $(DESTDIR)$(LIBDIR)/libzstd.a
-	$(RM) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT)
-	$(RM) $(DESTDIR)$(LIBDIR)/libzstd.$(SHARED_EXT_MAJOR)
+ifneq (,$(filter MSYS% MINGW%,$(UNAME)))
+	@$(RM) $(DESTDIR)$(BINDIR)/$(LIBZSTD)
+	@$(RM) $(DESTDIR)$(LIBDIR)/libzstd.dll.a
+else
+	$(RM) $(DESTDIR)$(DLLDIR)/libzstd.$(SHARED_EXT)
+	$(RM) $(DESTDIR)$(DLLDIR)/libzstd.$(SHARED_EXT_MAJOR)
+endif
 	$(RM) $(DESTDIR)$(LIBDIR)/$(LIBZSTD)
 	$(RM) $(DESTDIR)$(PKGCONFIGDIR)/libzstd.pc
 	$(RM) $(DESTDIR)$(INCLUDEDIR)/zstd.h

